### ISSUE TITLE
Can't log in after updating to PAM 1.5.0

### ISSUE THREAD

**Issue by ericonr** [linux-pam/linux-pam#277]
Hi! I'm looking into updating Void Linux's PAM package from 1.3.0 to 1.4.0. Due to shipping a musl libc version as well as a glibc one, it was necessary for me to add some patches to our build, both in order to get it to build at all, as well as fix the issue pointed out in #266 .

These patches can be seen here: https://github.com/void-linux/void-packages/pull/25506/commits/be4a3039fa87509326ca69bf2828f3602c4e5588

Unfortunately, after updating, I found myself unable not only to log into my system (through `agetty` and `login`), but also unable to unlock my lock screen ([swaylock](https://github.com/swaywm/swaylock)).

With the lockscreen, I can type in my password, but it simply fails validation. The error message is:

```
$ swaylock
2020-10-12 15:26:25 - [pam.c:101] pam_authenticate failed: unknown error (28)
```

And it repeats itself for every login attempt.

Interestingly, I can still gain privileges through `sudo` or [doas](https://github.com/Duncaen/OpenDoas).

If possible, I would certainly appreciate any help in debugging this.

**Comment by ericonr** [comment]
This is still happening with 1.5.0.

I can see it executes `/usr/bin/unix_chkpwd ericonr nullok`, but no other information.

**Comment by ericonr** [comment]
Turns out this was caused by having `pam_tally` in my config, which was deprecated in 1.4.0 and removed in 1.5.0, along with `pam_lastlog`, which stopped being built on musl somewhere along the line as well.

How could I keep up with these changes? Both are rather important ones... `pam_tally` was in NEWS and CHANGELOG, at least, but disabling `pam_lastlog` wasn't, and we had a very simple fix for it: https://github.com/void-linux/void-packages/blob/0e0c14e81fd882113dd6a90246130e1740b1a27c/srcpkgs/pam/patches/portability-fixes.patch

**Comment by ldv-alt** [comment]
> How could I keep up with these changes? Both are rather important ones... `pam_tally` was in NEWS and CHANGELOG, at least, but disabling `pam_lastlog` wasn't, and we had a very simple fix for it: https://github.com/void-linux/void-packages/blob/0e0c14e81fd882113dd6a90246130e1740b1a27c/srcpkgs/pam/patches/portability-fixes.patch

Sorry, it's the first time `pam_lastlog`  was ever mentioned, I suppose it hasn't been reported before.
In our tests everything is green including `pam_lastlog`.
If you have a specific problem with `pam_lastlog`, please report it in a regular way.
A reference to a strange looking patch somewhere in internet that reportedly fixes an unknown build issue certainly doesn't help.

**Comment by dilyn-corner** [comment]
It's possible you're running into [this](https://www.openwall.com/lists/musl/2013/12/03/3) old issue?

**Comment by ericonr** [comment]
Ok, the test for whether to build `pam_lastlog` or not was added in 73bf6d25ddb7a2cb73bceda3d880174b1d1e4a26 (which only landed in 1.4.0 - we are currently on 1.3); we had previously worked around the incompatibility issues with the patch I posted above, instead of not building the module at all. The commit that checks if `logwtmp` exists is probably more correct than our current workaround, so I guess all I could have asked for is that it be included in change logs.

However, it seems to work, for a certain value of that word. From a ssh session:

```
Last login: Wed Nov 25 22:06:44 2020 from 192.168.1.8
```

@ldv-alt 

> If you have a specific problem with pam_lastlog, please report it in a regular way.

If you think it's reasonable to have `pam_lastlog` working without `logwmp`, I can open a new issue for further discussion. Otherwise, the `pam_lastlog` removal happened so long ago that we might have been the only ones to notice, so there isn't much to act on anymore.

> A reference to a strange looking patch somewhere in internet that reportedly fixes an unknown build issue certainly doesn't help.

I try to upstream / discuss as many patches as I can, but not everyone has the same policy, unfortunately. The important part of the patch is https://github.com/void-linux/void-packages/blob/0e0c14e81fd882113dd6a90246130e1740b1a27c/srcpkgs/pam/patches/portability-fixes.patch#L1-L30 , where we remove the `logwtmp` calls based on the function's availability - so it's built on glibc and omitted on musl.
